### PR TITLE
Flink: Adds support for 1.18 version

### PR DESCRIPTION
**Pull request by rodmeneses**

### CODE REVIEW

File: flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java [hunk 1]
```diff
@@ -416,7 +416,7 @@ private static void assertAvroEquals(
         Collection<?> expectedArrayData = (Collection<?>) expected;
         ArrayData actualArrayData;
         try {
-          actualArrayData = (ArrayData) actual;
+          actualArrayData = convertToArray(actual);
```
Comment (stevenzwu) [inline flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:419]: is this a behavior change from 1.18? it seems like a breaking change to me.

Comment (rodmeneses) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:419]: only on 1.18 and it only affects the unit tests

Comment (stevenzwu) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:419]: I know you only updated the unit test. but it suggests that Flink `RowData` field has changed from `ArrayData` to the raw type array (like `int[]`). I am trying to understand if it has any impact on compatibility.

Comment (rodmeneses) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:419]: > I know you only updated the unit test. but it suggests that Flink `RowData` field has changed from `ArrayData` to the raw type array (like `int[]`). I am trying to understand if it has any impact on compatibility.

Thanks! cc: @pvary

Comment (pvary) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:419]: Based on @PrabhuJoseph's [comment](https://github.com/apache/iceberg/issues/8930#issuecomment-1806668299) the changes are caused by https://issues.apache.org/jira/browse/FLINK-33523.
Based on the jira, this is an expected general fix in Flink 1.18.

Comment (stevenzwu) [reply flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java:419]: With the clarify from the discussion thread that @pvary started, I am good with following up on this one separately when Flink 1.18.1 is released

**Review by stevenzwu — APPROVED** [review]

**Comment by stevenzwu** [comment]
thanks @rodmeneses for the contribution and @pvary for following up with the Flink community

**Comment by pvary** [comment]
@rodmeneses: Could you please update the docs too?

Here are the supported Flink versions: https://iceberg.apache.org/multi-engine-support/#apache-flink
And please check other places as well.
Thanks, Peter

**Comment by andrea-zanetti** [comment]
Hi, just to let you know that trying to download the runtime jar for the 1.18 version of Flink from [https://iceberg.apache.org/multi-engine-support/#apache-spark](https://iceberg.apache.org/multi-engine-support/#apache-spark) returns a 404.

**Comment by FranMorilloAWS** [comment]
@andrea-zanetti Could you share the proper link? i cant find the Jar Runtime for 1.18

**Comment by andrea-zanetti** [comment]
Hi @FranMorilloAWS,

I have the same exact problem: cannot find it on maven central, got here via google search but the download link is broken.

**Comment by rodmeneses** [comment]
> Hi @FranMorilloAWS,
> 
> I have the same exact problem: cannot find it on maven central, got here via google search but the download link is broken.
@andrea-zanetti
1.18 version will be available when Iceberg 1.5 is released

**Comment by pvary** [comment]
@andrea-zanetti, @FranMorilloAWS: Iceberg 1.5.0 will be the first version which supports Flink 1.18.x. It will be released soon, as the release process is already started

**Comment by andrea-zanetti** [comment]
Hi,

Thanks for you answer. I'll wait for the release of the new version.

**Pull request merged**